### PR TITLE
project-mongo-api Caroline Byström

### PR DESCRIPTION
**Pull request by Frivridna**
Finished the project.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,39 +1,187 @@
...
+    const oneDirector = await Director.findById(id)
+    if (oneDirector)  {
+      res.json({ data: oneDirector })
+    } else {
+      res.status(404).json({ error: 'Not found' })
+    }
+  } catch {
+    res.status(400).json({ error: 'Invalid request' })
+  }
+})
```
Comment (M4L5) [inline server.js:164]: I like that you used the try, catch 😎

**Review by M4L5 — COMMENTED** [review]
I like that you did several endpoints, the try/catch, also checked the database connection, added .env in the gitignore. You also used some things that I haven't read about which made me curious, for example Regex and .populate so I will look in to that :) Great job!

File: server.js [hunk 2]
```diff
@@ -1,39 +1,187 @@
...
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const titlesSchema = new mongoose.Schema({
+  show_id: String,
```
Comment (KarolCislowski) [inline server.js:16]: you will have _id field provided so you do not need to provide your own from the old dataset 🤓

File: server.js [hunk 3]
```diff
@@ -1,39 +1,187 @@
...
+  description: String
+})
+
+const directorsSchema = new mongoose.Schema({
+  name: String
+})
+
+const Title = mongoose.model('Title', titlesSchema)
+
+const Director = mongoose.model('Director', directorsSchema)
```
Comment (KarolCislowski) [inline server.js:39]: Nice that you provided 2 collections here 😸

File: server.js [hunk 4]
```diff
@@ -1,39 +1,187 @@
...
+}) 
+
+// return all titles
+app.get('/titles', async (req, res) => {
+    const titles = await Title.find().populate('director')
+    res.json({ length: titles.length, data: titles })
+})
+
+// query by year
+app.get('/titles/year', async (req, res) => {
```
Comment (KarolCislowski) [inline server.js:96]: when you use queries to filter your data, then do not create a separate endpoint. use generic GET endpoint for this collection: 

`/titles?year=2000` not `/titles/year?year=2000`

File: server.js [hunk 5]
```diff
@@ -1,39 +1,187 @@
...
+  
+  try {
+    const oneDirector = await Director.findById(id)
+    if (oneDirector)  {
+      res.json({ data: oneDirector })
+    } else {
+      res.status(404).json({ error: 'Not found' })
+    }
+  } catch {
+    res.status(400).json({ error: 'Invalid request' })
```
Comment (KarolCislowski) [inline server.js:162]: nice choice of error statuses 👍🏻

**Review by KarolCislowski — APPROVED** [review]
Nice API 👍🏻

**Pull request closed without merging**